runtime/runtime/src/lib.rs

```diff
@@ -1483,59 +1484,270 @@ impl Runtime {
+        total.add(outcome.gas_burnt, compute_usage)?;
+        metrics::TRANSACTION_PROCESSED_SUCCESSFULLY_TOTAL.inc();
+        Ok(outcome)
+    }
+
     /// Processes a collection of transactions.
     ///
     /// Fills the `processing_state` with local receipts generated during processing of the
     /// transactions.
     ///
-    /// Any transactions that fail to validate (e.g. invalid nonces, unknown signing keys,
-    /// insufficient NEAR balance, etc.) will be skipped, producing no receipts.
+    /// Depending on InvalidTxGenerateOutcomes feature status:
+    /// enabled => : Every transaction will result in an outcome registered in the processing state. Transaction that failed processing
+    /// will have `ExecutionStatus::Failure` status.
+    /// disabled => : Any transactions that fail to validate (e.g. invalid nonces, unknown signing keys,
+    /// insufficient NEAR balance, etc.) will be skipped, producing no receipts or outcomes
+    ///
+    /// In case processing any transaction results in `Err()` the chunk is dropped and
+    /// `RuntimeError` returned.
     fn process_transactions(
         &self,
         processing_state: &mut ApplyProcessingReceiptState,
-        signed_txs: SignedValidPeriodTransactions,
+        tx_vec: SignedValidPeriodTransactions,
         receipt_sink: &mut ReceiptSink,
     ) -> Result<(), RuntimeError> {
-        /// We track the transaction validity in a bit vector of this size. This type informs the
-        /// maximum size of the transactions' chunk processed with each rayon job.
-        type ValidBitmask = u32;
-        /// Avoid the overhead of inter-thread scheduling by processing at least this many
-        /// transactions for each instance of this overhead. This can reduce the number of
-        /// transaction chunks for smaller lists of transactions, however.
-        const MIN_CHUNK_SIZE: usize = 8;
-        /// Avoid splitting transactions into just $NUM_THREADS chunks, as that can result in an
-        /// increased tail latency when one of the threads is slower at processing its chunk
-        /// compared to others (whatever reason may be for that.) Splitting into smaller chunks
-        /// allows the load to be distributed across threads more evenly and any tail latency
-        /// reduced due to the last chunk(s) being smaller.
-        const TARGET_CHUNKS_PER_THREAD: usize = 4;
-        let total = &mut processing_state.total;
         let apply_state = &mut processing_state.apply_state;
-        let state_update = &mut processing_state.state_update;
-        let tx_vec = signed_txs.into_nonexpired_transactions();
+        let state_update = &processing_state.state_update;
         let len = tx_vec.len();
-        let chunk_count_target = rayon::current_num_threads() * TARGET_CHUNKS_PER_THREAD;
-        let chunk_size =
-            (tx_vec.len() / chunk_count_target).clamp(MIN_CHUNK_SIZE, ValidBitmask::BITS as _);
+        const CHUNK_SIZE: usize = 32;
```

Comment:
This discards the entire chunk sizing logic. This specific number may still work alright for our benchmark scenarios, but real workloads will more likely than not just run all the work in a single chunk.